apis/condition_set.go

```diff
@@ -207,6 +210,78 @@ func (r conditionsImpl) severity(t ConditionType) ConditionSeverity {
 	return ConditionSeverityInfo
 }
 
+// RemoveCondition removes the condition that matches the ConditionType
+func (r conditionsImpl) RemoveCondition(t ConditionType) {
+	var conditions Conditions
+
+	if r.accessor == nil {
+		return
+	}
+	cond := r.GetCondition(t)
+	if cond == nil {
+		return
+	}
+	for _, c := range r.accessor.GetConditions() {
+		if c.Type != t {
+			conditions = append(conditions, c)
+		}
+	}
+
+	// Sorted for convenience of the consumer, i.e. kubectl.
+	sort.Slice(conditions, func(i, j int) bool { return conditions[i].Type < conditions[j].Type })
+	r.accessor.SetConditions(conditions)
+
+	happy := r.GetCondition(r.happy)
+	if r.isTerminal(t) {
+		if cond.IsFalse() && happy.IsFalse() {
```

Comment:
I think all of this logic can be reduced down to resetting all of the dependents to their current state with the mark methods without regard to the happy condition with testing that there are dependents. If you removed the only dependent, I would assume happy goes from false to unknown, or stays the same? 

So: ignore the happy condition if there are deps, for each dep, set it again using the `Mark` methods. This will cause `happy` updates to be propagated.